internal/idler/user_idler.go

```diff
@@ -103,10 +103,34 @@ func (idler *UserIdler) checkIdle() error {
 			return err
 		}
 		if enabled {
-			idler.doIdle()
+			if err := idler.doIdle(); err != nil {
+				idler.user.IdleStatus = model.IdleStatus{
```

Comment:
I see lot of repeating code. how about  

```go
idler.user.IdlerStatus = model.IdleStatusFromError(err)
// ... 
func IdleStatusFromrError(err error) IdleStatus {
 ...
  return status
}
```

where `IdleStatusFromrError` returns IdleStatus.`Success` to `true/false` based on the `err` being `nil` or not.

Reply:
btw, Do we not need any test for this?

Reply:
@sthaha changed the implementation, PTAL